src/assets/src/hooks/useGoogleAnalytics.ts

```diff
@@ -1,22 +1,72 @@
 import { useState, useEffect } from 'react';
 import GoogleAnalytics from 'react-ga4';
 import { useLocation } from 'react-router-dom';
+import { useOneTrust } from './useOneTrust';
+
 
 export const useGoogleAnalytics = (googleAnalyticsId?: string, debug?: boolean) => {
     let location = useLocation();
+
     const [initialized, setInitialized] = useState(false);
     const [previousPage, setPreviousPage] = useState(null as string | null);
 
     if (googleAnalyticsId && !initialized) {
+        GoogleAnalytics.gtag("consent", "default", {
+            ad_storage: "denied",
+            analytics_storage: "denied",
+            functionality_storage: "denied",
+            personalization_storage: "denied",
+            ad_user_data: "denied",
+            ad_personalization: "denied",
+            wait_for_update: 500
+        });
         setInitialized(true);
         GoogleAnalytics.initialize(googleAnalyticsId, { testMode: debug });
     }
 
+    const [loaded, oneTrustActiveGroups, setOptanonWrapper] = useOneTrust();
+    if (loaded) {
+        console.log("loaded oneTrustActiveGroups" + oneTrustActiveGroups);
+    }
+
     useEffect(() => {
         const page = location.pathname + location.search + location.hash;
         if (googleAnalyticsId && page !== previousPage) {
             setPreviousPage(page);
             GoogleAnalytics.send({ hitType: "pageview", page });
         }
     }, [location]);
+    
+    useEffect(() => {
+        const updateGtagConsent = () => {
+
+            if (oneTrustActiveGroups.includes("C0002")) {
+              GoogleAnalytics.gtag("consent", "update", { analytics_storage: "granted" });
+            }
+            if (oneTrustActiveGroups.includes("C0003")) {
+              GoogleAnalytics.gtag("consent", "update", { functional_storage: "granted" });
+            }
+            if (oneTrustActiveGroups.includes("C0004")) {
```

Comment:
If you add some links or info defining what these code are  `C0002` `C0003' `C0004` would be useful. looks like magic alpha-numeric series.

Reply:
I think you can add this link https://vpcomm.umich.edu/resources/cookie-disclosure/ to the file for more details. 
Could benefit from one-line commets

Strictly Necessary Cookies: C0003 (Functional)

Reply:
added an enum and some clarifying comments